classes/seb_cipher.php

```diff
@@ -0,0 +1,119 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Class cipher is for encrypting and decrypting SEB config files.
+ *
+ * @package    quizaccess_seb
+ * @copyright  2018 Nicholas Hoobin <nicholashoobin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace quizaccess_seb;
+
+use RNCryptor\RNCryptor\Cryptor;
+use RNCryptor\RNCryptor\Decryptor;
+use RNCryptor\RNCryptor\Encryptor;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../vendor/autoload.php');
+
+class seb_cipher {
+
+    /**
+     * Encrypt a seb file with a password ready for use as a seb exam link.
+     *
+     * @param string $xml
+     * @param string $password
+     * @return string
+     * @throws \Exception
+     */
+    public static function encrypt(string $xml, string $password) : string {
+        $cryptor = new Encryptor;
+
+        // These replacements are for converting PList to XML representation.
+        $replacements = [
+            '<array />'         => '<array></array>',
+            '<dict />'          => '<dict></dict>',
+            '<data />'          => '<data></data>',
+            '<true />'          => '<true/>',
+            '<false />'         => '<false/>',
+            "\r\n"              => "\n",       // Convert DOS to UNIX line endings.
+        ];
+
+        foreach ($replacements as $key => $replacement) {
+            $xml = str_replace($key, $replacement, $xml);
+        }
+
+        // Stripe the trailing newline.
+        $data = trim($xml);
+
+        // First pass compression of encrypted data.
+        $data = gzencode($data);
```

Comment:
Not sure if I understand why we do that

Reply:
I've added detailed descriptions of process in method docs and references to steps in code comments.